libs/testproxy/server.go

```diff
@@ -0,0 +1,177 @@
+package testproxy
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/databricks/cli/internal/testutil"
+	"github.com/databricks/cli/libs/testserver"
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/client"
+	"github.com/databricks/databricks-sdk-go/config"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type ProxyServer struct {
+	*httptest.Server
+
+	t      testutil.TestingT
+	router *mux.Router
+
+	apiClient        *client.DatabricksClient
+	RequestCallback  func(request *testserver.Request)
+	ResponseCallback func(request *testserver.Request, response *testserver.EncodedResponse)
+}
+
+// This creates a reverse proxy server that sits in front of a real Databricks
+// workspace. This is useful for recording API requests and responses in
+// integration tests.
+//
+// Note: We cannot simply proxy the request from a localhost URL to a real
+// workspace. This is because auth resolution in the Databricks SDK relies
+// what the URL actually looks like to determine the auth method to use.
+// For example, in OAuth flows, the SDK can make requests to different Microsoft
+// OAuth endpoints based on the nature of the URL.
+// For reference, see:
+// https://github.com/databricks/databricks-sdk-go/blob/79e4b3a6e9b0b7dcb1af9ad4025deb447b01d933/common/environment/environments.go#L57
+func New(t testutil.TestingT) *ProxyServer {
+	router := mux.NewRouter()
+	server := httptest.NewServer(router)
```

Comment:
Why have a dependency on gorilla/mux or have a router at all?

Can't we just http.Handler to  https://pkg.go.dev/net/http/httptest#NewServer ?

Reply:
We use gorilla/mux in the testserver. I'd prefer to use the same package here. It keeps cognitive overhead lower when working on these.

Reply:
I'm aware but there is a use case for it in testserver, not relevant here.

Reply:
using httptest now.